### PR TITLE
Rewrite MemoryMappedFiles tests

### LINKED ISSUE #928: Rewrite System.IO.MemoryMappedFiles tests
According to the current code coverage report, this library is only at 77%.  Our general goal is closer to 90%, though this will depend on the library.  We should investigate to see whether the current coverage level is appropriate, and if not, improve our test suites to provide better coverage.

### PR DESCRIPTION
**Pull request by stephentoub**
More details are in each commit's description, but at a high-level this PR completely rewrites the memory-mapped files tests.  Along the way I found a few bugs in the Unix implementation, and this PR fixes those.  I also found some dead code, which this PR removes.

Fixes #928, #1061, #560

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @ianhays, @ellismg, @ericstj, @FiveTimesTheFun

**Comment by ellismg** [comment]
Thanks Steve!  I did a pass over the tests and left some feedback, nothing major.  Might be good to wait for another set of eyes before merging however.

**Comment by stephentoub** [comment]
Thanks, Matt!  I addressed your feedback.

File: src/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateFromFile.Tests.cs [hunk 1]
```diff
@@ -0,0 +1,810 @@
...
+        /// </summary>
+        [Theory]
+        [InlineData(FileAccess.ReadWrite, MemoryMappedFileAccess.Read)]
+        [InlineData(FileAccess.ReadWrite, MemoryMappedFileAccess.ReadWrite)]
+        [InlineData(FileAccess.ReadWrite, MemoryMappedFileAccess.CopyOnWrite)]
+        [InlineData(FileAccess.Read, MemoryMappedFileAccess.Read)]
+        [InlineData(FileAccess.Read, MemoryMappedFileAccess.CopyOnWrite)]
+        public void FileAccessAndMapAccessCombinations_Valid(FileAccess fileAccess, MemoryMappedFileAccess mmfAccess)
+        {
+            const int Capacity = 4096;
```
Comment (ianhays) [inline src/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateFromFile.Tests.cs]: You've got Capacity separately declared in multiple locations as 4096. Might be better to just move it up with a name like DefaultCapacity so it's only declared in one place.

File: src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+        [Theory]
+        [InlineData(0, 8192)]
+        [InlineData(8100, 92)]
+        [InlineData(0, 20)]
+        [InlineData(1, 8191)]
+        [InlineData(17, 8175)]
+        [InlineData(17, 20)]
+        public void AllReadWriteMethods(long offset, long size)
+        {
+            foreach (MemoryMappedFile mmf in CreateSampleMaps(8192))
```
Comment (ianhays) [inline src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs]: CreateSampleMaps(8192) fine, or should that number instead be "size"?

Comment (stephentoub) [inline src/System.IO.MemoryMappedFiles/tests/MemoryMappedViewAccessor.Tests.cs]: Good question, but I actually wanted this to be 8192, a value at least as large as any of the offset+size combinations.  I could have also made it offset+size, but part of what I wanted tested was creating accessors that are small slices of the overall map.

**Comment by ianhays** [comment]
Just gave this a look over and found no egregious errors :) Looks good!

File: src/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateFromFile.Tests.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateFromFile.Tests.cs]: I went back and forth on that.  DefaultCapacity means something specific in this library, 0 being passed to mean "use the size of the underlying file or map", so I didn't want to use that, and I couldn't come up with something better.  Plus, my choices in many of these cases were largely random, any value would do as long as it was consistent within the method, hence my defining a constant in each of the methods where I needed consistency within that method.  I think I'll leave it this way for now... we can always choose to unify them later if desired.

**Comment by stephentoub** [comment]
Thanks for the reviews, guys!

**Pull request merged**